sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuKeyBatchingIterator.scala

```diff
@@ -115,100 +117,119 @@ class GpuKeyBatchingIterator(
     }
   }
 
-  private def concatPending(last: Option[Table] = None): ColumnarBatch = {
-    var peak = 0L
-    try {
-      withResource(new NvtxWithMetrics("concat pending", NvtxColor.CYAN, concatTime)) { _ =>
-        withResource(mutable.ArrayBuffer[Table]()) { toConcat =>
-          while (pending.nonEmpty) {
-            withResource(pending.dequeue()) { spillable =>
+  private def concatPending(last: Option[SpillableColumnarBatch] = None): ColumnarBatch = {
+    val spillableBuffers = new ArrayBuffer[SpillableColumnarBatch]()
+    while (pending.nonEmpty) {
+      spillableBuffers.append(pending.dequeue())
+    }
+    pendingSize = 0
+    last.foreach { lastSpill =>
```

Comment:
nit, could be replaced with: `spillableBuffers.appendAll(last)`. I didn't know that you could, but it looks like `appendAll` can handle an `Option` and if set to `None` it is a noop.